joblib/_memmapping_reducer.py

```diff
@@ -427,7 +510,7 @@ def _cleanup():
         # to avoid confusing the caller and make it tricky to collect the
         # temporary folder
         backward_reduce_ndarray = ArrayMemmapReducer(
-            None, pool_folder, mmap_mode, verbose)
+            None, pool_folder, mmap_mode, track_memmap_in_child, verbose)
```

Comment:
Backward reducers should never send back temporary numpy arrays so this flag can always be set to False here.